src/main/scala/com/bazaarvoice/sswf/service/WorkflowManagement.scala

```diff
@@ -0,0 +1,264 @@
+    )
+
+    WorkflowExecution(workflowId, run.getRunId)
+  }
+
+
+  /**
+   * List all the executions for this domain and workflow within the time window. (regardless of version)
+   * @param from Start time to search
+   * @param to End time to search
+   * @return an unmodifiable list of matching executions
+   */
+  def listExecutions(from: Date, to: Date): java.util.List[WorkflowExecutionInfo] = {
+    val openRequest: ListOpenWorkflowExecutionsRequest = new ListOpenWorkflowExecutionsRequest()
+       .withDomain(domain)
+       .withTypeFilter(new WorkflowTypeFilter().withName(workflow))
+       .withStartTimeFilter(new ExecutionTimeFilter().withLatestDate(to).withOldestDate(from))
+
+    val closedRequest: ListClosedWorkflowExecutionsRequest = new ListClosedWorkflowExecutionsRequest()
+       .withDomain(domain)
+       .withTypeFilter(new WorkflowTypeFilter().withName(workflow))
+       .withStartTimeFilter(new ExecutionTimeFilter().withLatestDate(to).withOldestDate(from))
+
+    innerListExecutions(openRequest, closedRequest)
+  }
+
+  /**
+   * List all the executions for this domain, workflow, and workflowId within the time window. (regardless of version)
+   *
+   * @param from Start time to search
+   * @param to End time to search
+   * @param workflowId The particular workflow id to list executions for
+   * @return an unmodifiable list of matching executions
+   */
+  def listExecutions(from: Date, to: Date, workflowId: String): java.util.List[WorkflowExecutionInfo] = {
+    val openRequest: ListOpenWorkflowExecutionsRequest = new ListOpenWorkflowExecutionsRequest()
+       .withDomain(domain)
+       .withTypeFilter(new WorkflowTypeFilter().withName(workflow))
+       .withExecutionFilter(new WorkflowExecutionFilter().withWorkflowId(workflowId))
+       .withStartTimeFilter(new ExecutionTimeFilter().withLatestDate(to).withOldestDate(from))
+
+    val closedRequest: ListClosedWorkflowExecutionsRequest = new ListClosedWorkflowExecutionsRequest()
+       .withDomain(domain)
+       .withTypeFilter(new WorkflowTypeFilter().withName(workflow))
+       .withExecutionFilter(new WorkflowExecutionFilter().withWorkflowId(workflowId))
+       .withStartTimeFilter(new ExecutionTimeFilter().withLatestDate(to).withOldestDate(from))
+
+    innerListExecutions(openRequest, closedRequest)
+  }
+
+  /**
+   * List all the events from an execution of a workflow.
+   * @param workflowId The particular workflow id to list events for
+   * @param runId The particular run of the workflow id to list events for
+   * @return
+   */
+  def describeExecution(workflowId: String, runId: String): StepsHistory[SSWFInput, StepEnum] = {
+    val request: GetWorkflowExecutionHistoryRequest = new GetWorkflowExecutionHistoryRequest()
+       .withDomain(domain)
+       .withExecution(new model.WorkflowExecution().withWorkflowId(workflowId).withRunId(runId))
+
+    val iterateFn = (prev: History) => {
+      if (prev == null | prev.getNextPageToken == null) null
```

Comment:
this will throw an NPE if `prev` is null. I think you want `||` since `|` will evaluate both conditions.